### ISSUE TITLE
Hybrid solver prone to slow convergence

### ISSUE THREAD

**Issue by radarsat1** [siconos/siconos#214]
I've noticed lately that Siconos seems slower than it used to, and in fact I find it strange that some simulations tend to max out the iterations even when the object is motionless, for example, if you run,

    cd examples/Mechanics/GeometricPrimitives
    python cube.py

which has very high precision settings for some reason (itermax=100000, h=0.0005, Newton_max_iter=20, tolerance=1e-8), you will see that around step 4000 and above, when the cube has come to a stop, the output is,

```
step 4381 of 20000
TimeStepping::DefaultCheckSolverOutput:
Non smooth solver warning : output message from numerics solver is equal to 1
output in hdf5 file at step  4381
number of contacts 3
SolverInfos at time : 2.1905 iterations=  100000 precision= 1.18097972381e-08

step 4382 of 20000
TimeStepping::DefaultCheckSolverOutput:
Non smooth solver warning : output message from numerics solver is equal to 1
output in hdf5 file at step  4382
number of contacts 3
SolverInfos at time : 2.191 iterations=  100000 precision= 1.18099657438e-08
```

etc.  The contacts are *not changing*.  The solutions change only slightly between steps.  I have printed out the lambda vectors and the solution only changes by about 1e-4, and yet it takes 100000 iterations to get the precision down to 1e-8, at every step.

I verified a few things:
* contacts are preserved between steps, not generated new
* lambda values are copied correctly between the Interactions and the OSNSP

So then I started playing with the default settings for numerics, and I found what had impact was changing

    fcOptions.solverId = Numerics.SICONOS_FRICTION_3D_ONECONTACT_NSN_GP_HYBRID

to

    fcOptions.solverId = Numerics.SICONOS_FRICTION_3D_ONECONTACT_NSN_GP

Now the number of iterations when the cube is stable average around 2 or 3.
With such a high maxiter, the difference in total simulation time really adds up.  Time for cube.py with NSN_GP_HYBRID:
```
$ time python3 cube.py 2>&1 >/dev/null

real	43m52,463s
user	52m49,208s
sys	52m3,040s
```

very long for just a cube simulation!  Compared with if I change to NSN_GP,

```
$ time python3 cube.py 2>&1 >/dev/null

real	0m41,067s
user	0m52,516s
sys	1m9,864s
```

**Comment by xhub** [comment]
Thanks for reporting that. Let's try to save some of those examples and create a benchmark dataset.

**Comment by vacary** [comment]
The hybrid was slow for two main reasons:
- the measure of error between PLI solver and NSN solver is not the same. Therefore, an improvement of the residual in the NSN method may lead to an increase of the absolute error and finally to destabilize the local solver
- by default, the solver starts by the PLI solver and then may deteriorate the quality of the solution.

In e140907d22067b75672ff256bb37995f385119f1, I tried to fix this problem. With NSN_GP_HYBRID, we get similar performance as for NSN_GP on the simple cube example.

I propose to close the issue. Feel free to reopen if needed

**Comment by vacary** [comment]
Finally, I found another bug in the hybrid solver. The value of the rho parameter in the PLI approach was not initialized leading to possibly negative rho and nan results in the PLI algorithm.

Rather than stabilizing the solution, the PLI  iterations was destabilizing the local solver !!!

This is fixed in 01b5424d389e6b0724b652766a6f397dc8334db0
